core/src/test/java/org/opentosca/toscana/plugins/kubernetes/KubernetesPluginTest.java

```diff
@@ -0,0 +1,96 @@
+
+import org.opentosca.toscana.core.BaseSpringTest;
+import org.opentosca.toscana.core.csar.Csar;
+import org.opentosca.toscana.core.parse.CsarParseService;
+import org.opentosca.toscana.core.parse.InvalidCsarException;
+import org.opentosca.toscana.core.testdata.TestCsars;
+import org.opentosca.toscana.core.testdata.TestPlugins;
+import org.opentosca.toscana.core.testdata.TestTransformationContext;
+import org.opentosca.toscana.core.transformation.TransformationContext;
+import org.opentosca.toscana.core.util.Preferences;
+import org.opentosca.toscana.plugins.model.InvalidDockerAppException;
+
+import org.assertj.core.util.Lists;
+import org.eclipse.winery.model.tosca.yaml.TServiceTemplate;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.opentosca.toscana.core.testdata.TestCsars.CSAR_YAML_VALID_DOCKER_SIMPLETASK;
+
+public class KubernetesPluginTest extends BaseSpringTest {
+    private static final Logger log = LoggerFactory.getLogger(KubernetesPluginTest.class);
+
+    private static TServiceTemplate tServiceTemplate;
+    private static KubernetesPlugin plugin;
+    @Autowired
+    private CsarParseService csarParser;
+    @Autowired
+    private Preferences preferences;
+    @Autowired
+    private TestTransformationContext testTransformationContext;
+    @Autowired
+    private TestCsars testCsars;
+
+    @Before
+    public void setUp() {
+        plugin = new KubernetesPlugin();
+    }
+
+    @Test
+    public void validTServiceTemplate() throws InvalidDockerAppException, FileNotFoundException, InvalidCsarException {
+        Csar csar = testCsars.getCsar(CSAR_YAML_VALID_DOCKER_SIMPLETASK);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test(expected = InvalidDockerAppException.class)
+    public void dockerAppMissing() throws FileNotFoundException, InvalidCsarException, InvalidDockerAppException {
+        Csar csar = testCsars.getCsar(TestCsars.CSAR_YAML_INVALID_DOCKERAPP_MISSING);
+        tServiceTemplate = csarParser.parse(csar);
+        plugin.getDockerApp(tServiceTemplate);
+    }
+
+    @Test
+    public void transformTest() throws Exception {
+        File artifactRootPath = new File(preferences.getDataDir(), "test-csar/transformations/testplatform_one/");
+        TransformationContext transformationContext = testTransformationContext.getContext(CSAR_YAML_VALID_DOCKER_SIMPLETASK, TestPlugins.PLATFORM1);
+        assertNotNull(transformationContext);
+        plugin.transform(transformationContext);
+
+        checkFilesInTransformationRoot(artifactRootPath);
+        checkDockerFilesInTransformationRoot(artifactRootPath);
+    }
+
+    private void checkDockerFilesInTransformationRoot(File artifactRootPath) {
```

Comment:
Look, this is a perfect example: Yesterday, the TransformationDao implementation changed.
Transformation Artifacts (not the resulting archive, but the single files) are now copied to the subdir /contents in the transformationRootDir. If you need the paths, please use transformationDao.getContentDir() or transformationDao.getRootDir() (here you will find the .zip)